pkg/store/gcs.go

```diff
@@ -262,11 +274,26 @@ func (r *seriesResult) response() ([]storepb.Series, error) {
 			Chunks: make([]storepb.Chunk, 0, len(e.chunks)),
 		}
 		for _, l := range e.lset {
+			// Skip if the external labels of the block overrule the series' label.
+			// NOTE(fabxc): maybe move it to a prefixed version to still ensure uniqueness of series?
```

Comment:
I vote for adding these... similar to how kubernetes adds its meta labels (eg `__meta_kubernetes_pod_node_name`)

Reply:
In Prometheus federation we usually add an `exported_` prefix. But we can discuss other options as well.